lib/pure/parsecsv.nim

```diff
@@ -162,9 +173,52 @@ proc readRow*(my: var CsvParser, columns = 0): bool =
           $col & " columns")
   inc(my.currRow)
 
+proc tryReadRow*(my: var CsvParser, columns = 0): bool =
+  ## reads the next row without raising `CsvError`;
+  ## Returns false if the end of the file
+  ## has been encountered else true.
+  ##
+  ## Blank lines are skipped.
+  var col = 0 # current column
+  var oldpos = my.bufpos
+  while my.buf[my.bufpos] != '\0':
+    var oldlen = my.row.len
+    if oldlen < col+1:
+      setLen(my.row, col+1)
+      my.row[col] = ""
+    parseField(my, my.row[col])
+    inc(col)
+    if my.buf[my.bufpos] == my.sep:
+      inc(my.bufpos)
+    else:
+      case my.buf[my.bufpos]
+      of '\c', '\l':
+        # skip empty lines:
+        while true:
+          case my.buf[my.bufpos]
+          of '\c': my.bufpos = handleCR(my, my.bufpos)
+          of '\l': my.bufpos = handleLF(my, my.bufpos)
+          else: break
+      of '\0': discard
+      else: discard  # no delimeter - ignore
+      break
+
+  setLen(my.row, col)
+  result = col > 0
+  inc(my.currRow)
+
+proc readRow*(my: var CsvParser, filename: string,
+           separator = ',', quote = '"', escape = '\0',
+           skipInitialSpace = false, columns = 0): bool =
+  ## Opens the file, then call the generic `tryReadRow()`
+  if my.filename.isNil:
+    my.open(filename, separator, quote, escape, skipInitialSpace)
+  result = my.tryReadRow(columns = columns)
+
 proc close*(my: var CsvParser) {.inline.} =
   ## closes the parser `my` and its associated input stream.
   lexbase.close(my)
+  my.filename = nil
```

Comment:
First time, yes it is nil.
If opening a second time using the new readRow(), then need to identify that it isn't already open.
Alternately, can add a flag to CsvParser.